smack-integration-test/src/main/java/org/jivesoftware/smackx/ox/OXSecretKeyBackupIntegrationTest.java

```diff
@@ -122,38 +123,45 @@ public void test() throws InvalidAlgorithmParameterException, NoSuchAlgorithmExc
+            assertTrue(self.getSecretKeys().contains(keyFingerprint.getKeyId()));
 
-        PGPSecretKeyRing beforeSec = beforeStore.getSecretKeyRing(alice, keyFingerprint);
-        assertNotNull(beforeSec);
+            PGPSecretKeyRing beforeSec = beforeStore.getSecretKeyRing(alice, keyFingerprint);
+            assertNotNull(beforeSec);
 
-        PGPPublicKeyRing beforePub = beforeStore.getPublicKeyRing(alice, keyFingerprint);
-        assertNotNull(beforePub);
+            PGPPublicKeyRing beforePub = beforeStore.getPublicKeyRing(alice, keyFingerprint);
+            assertNotNull(beforePub);
 
-        OpenPgpSecretKeyBackupPassphrase backupPassphrase =
+            OpenPgpSecretKeyBackupPassphrase backupPassphrase =
                 openPgpManager.backupSecretKeyToServer(availableSecretKeys -> availableSecretKeys);
 
-        FileBasedOpenPgpStore afterStore = new FileBasedOpenPgpStore(afterPath);
-        afterStore.setKeyRingProtector(new UnprotectedKeysProtector());
-        PainlessOpenPgpProvider afterProvider = new PainlessOpenPgpProvider(afterStore);
-        openPgpManager.setOpenPgpProvider(afterProvider);
+            FileBasedOpenPgpStore afterStore = new FileBasedOpenPgpStore(afterPath);
+            afterStore.setKeyRingProtector(new UnprotectedKeysProtector());
+            PainlessOpenPgpProvider afterProvider = new PainlessOpenPgpProvider(afterStore);
+            openPgpManager.setOpenPgpProvider(afterProvider);
 
-        OpenPgpV4Fingerprint fingerprint = openPgpManager.restoreSecretKeyServerBackup(() -> backupPassphrase);
+            OpenPgpV4Fingerprint fingerprint = openPgpManager.restoreSecretKeyServerBackup(() -> backupPassphrase);
 
-        assertEquals(keyFingerprint, fingerprint);
+            assertEquals(keyFingerprint, fingerprint);
 
-        assertTrue(self.getSecretKeys().contains(keyFingerprint.getKeyId()));
+            assertTrue(self.getSecretKeys().contains(keyFingerprint.getKeyId()));
 
-        assertEquals(keyFingerprint, self.getSigningKeyFingerprint());
+            assertEquals(keyFingerprint, self.getSigningKeyFingerprint());
 
-        PGPSecretKeyRing afterSec = afterStore.getSecretKeyRing(alice, keyFingerprint);
-        assertNotNull(afterSec);
-        assertArrayEquals(beforeSec.getEncoded(), afterSec.getEncoded());
+            PGPSecretKeyRing afterSec = afterStore.getSecretKeyRing(alice, keyFingerprint);
+            assertNotNull(afterSec);
+            assertArrayEquals(beforeSec.getEncoded(), afterSec.getEncoded());
 
-        PGPPublicKeyRing afterPub = afterStore.getPublicKeyRing(alice, keyFingerprint);
-        assertNotNull(afterPub);
-        assertArrayEquals(beforePub.getEncoded(), afterPub.getEncoded());
+            PGPPublicKeyRing afterPub = afterStore.getPublicKeyRing(alice, keyFingerprint);
+            assertNotNull(afterPub);
+            assertArrayEquals(beforePub.getEncoded(), afterPub.getEncoded());
+        } finally {
+            Thread.sleep(10000);
```

Comment:
Could those `Thread.sleep()`s be dropped?

Reply:
:facepalm: